services/api-server/src/simcore_service_api_server/models/schemas/solvers.py

```diff
@@ -0,0 +1,170 @@
+import re
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional, Union
+from uuid import UUID
+
+from pydantic import BaseModel, Field, HttpUrl, conint, constr
+
+LATEST_VERSION = "latest"
+KEY_RE = r"^(simcore)/(services)/(comp)(/[^\s/]+)+$"  # NOTE: needs to end with / !!
+
+
+# Human-readable unique identifier
+KeyIdentifier = constr(strip_whitespace=True, min_length=3)
+SolverImageName = constr(regex=KEY_RE, strip_whitespace=True)
+
+
+class SolverImage(BaseModel):
+    # This is an image. Notice that tags refer to this image
+    uuid: UUID = Field(..., description="Image sha256 unique identifier")
+    name: SolverImageName = Field(
+        ...,
+        description="Name of the solver image including namespace and excluding tag",
+    )
+    maintainer: str
+    released: datetime
+
+
+class Solver(BaseModel):
+    """A released solver with a specific version
+
+    This version might have human-readable alias (e.g. latest) or
+    hierarchical version tags (e.g. 3, 3.2)
+    """
+
+    uuid: UUID = Field(..., description="Same as the solver's image sha256")
+    name: str = Field(..., description="Image name including namespace")
+    version: str  # complete tag.  e.g. 3.4.5 TODO: regex for version in python PEP
+    version_aliases: List[str] = []  # remaining tags
+
+    title: str
+    description: Optional[str]
+    maintainer: str
+    released: Optional[datetime]  # TODO: turn into required
+
+    solver_url: HttpUrl
+
+    @classmethod
+    def create_from_image(cls, img: SolverImage, tags: List, **kwargs) -> "Solver":
+        version = None
+        alias = []
+        for tag in tags:
+            if re.match(r"\d+\.\d+\.\d+", tag):
+                version = tag
+            else:
+                alias.append(tag)
+
+        return cls(
+            uuid=img.uuid,
+            name=img.name,
+            title=img.name.split("/")[-1],
+            maintainer=img.maintainer,
+            released=img.released,
+            version=version,
+            version_aliases=alias,
+            **kwargs,
+        )
+
+
+class Job(BaseModel):
+    job_id: UUID
+    inputs_sha: str
+    solver_id: UUID
+
+    # hyperlinks
+    solver_url: HttpUrl
+    inspect_url: HttpUrl
+    outputs_url: HttpUrl
+
+
+# TODO: these need to be in sync with celery task states
+class TaskStates(str, Enum):
+    UNDEFINED = "undefined"
+    PENDING = "pending"
+    RUNNING = "running"
+    SUCCESS = "success"
+    FAILED = "failed"
```

Comment:
right so. this needs to be not in sync, but defined in models_library. if it's not there let's move it over.

Reply:
As below